code/stable/projectile/SRS/HTML/Projectile_SRS.html

```diff
@@ -459,6 +467,71 @@ <h2>Organization of Document</h2>
         </div>
       </div>
     </div>
+    <div id="Sec:GenSysDesc">
+      <div class="section">
+        <h1>General System Description</h1>
+        <p class="paragraph">
+          This section provides general information about the system. It identifies the interfaces between the system and its environment, describes the user characteristics, and lists the system constraints.
+        </p>
+        <div id="Sec:SysContext">
+          <div class="subsection">
+            <h2>System Context</h2>
+            <p class="paragraph">
+              <a href=#Figure:sysCtxDiag>Fig:sysCtxDiag</a> shows the system context. A circle represents an entity external to the software, the user in this case. A rectangle represents the software system itself (Projectile). Arrows are used to show the data flow between the system and its environment.
+            </p>
+            <div id="Figure:sysCtxDiag">
+              <figure>
+                <img src="../../../../datafiles/projectile/SystemContextFigure.png" alt="System Context" >
+                <figcaption>System Context</figcaption>
+              </figure>
+            </div>
+            <p class="paragraph">
+              The interaction between the product and the user is through an application programming interface. The responsibilities of the user and the system are as follows:
+            </p>
+            <ul class="list">
+              <li>
+                User Responsibilities
+                <ul class="list">
+                  <li>
+                    Provide initial conditions of the physical state of the motion and the input data related to the Projectile, ensuring no errors in the data entry.
+                  </li>
+                  <li>Ensure that consistent units are used for input variables.</li>
+                  <li>
+                    Ensure required <a href=#Sec:Assumps>software assumptions</a> are appropriate for any particular problem input to the software.
+                  </li>
+                </ul>
+              </li>
+              <li>
+                Projectile Responsibilities
+                <ul class="list">
+                  <li>
+                    Detect data type mismatch, such as a string of characters input instead of a floating point number.
```

Comment:
This will have interesting consequences for my testing project, since I don't think Projectile's code actually does this!

Reply:
Although there is no code written to detect the type mismatch, Projectile does technically do this, since it will cause a run time error from the interpreter or compiler.  Having said that, if we had a language that wasn't typed, we could have a problem.  Even in languages with typing, it won't be a very graceful failure.

Reply:
Should this line be removed? Because as said above, this is similar to what is said in other examples.

Reply:
If other examples say it, my guess is that they don't actually implement this in the code, either. My guess is that this will come up during my project, so my vote is to leave this in for now and wait for it to be handled later

Reply:
Yes, leave it in for now.  It is a good responsibility for the code to take on.  If we don't do it in an elegant way, we should address that (eventually), rather than change the requirements.